org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/internal/utils/MarkerUtils.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.tm4e.ui.internal.utils;
+
+import static org.eclipse.tm4e.core.internal.utils.NullSafetyHelper.*;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.tm4e.core.model.ITMModel;
+import org.eclipse.tm4e.core.model.ModelTokensChangedEvent;
+import org.eclipse.tm4e.core.model.TMToken;
+import org.eclipse.tm4e.ui.TMUIPlugin;
+import org.eclipse.tm4e.ui.internal.model.TMDocumentModel;
+
+public final class MarkerUtils {
+
+	private abstract static class AbstractMarkerConfig {
+		final String type;
+		final int priority;
+		final int severity;
+
+		AbstractMarkerConfig(final String type, final int priority, final int severity) {
+			this.type = type;
+			this.priority = priority;
+			this.severity = severity;
+		}
+	}
+
+	private static final class ProblemMarkerConfig extends AbstractMarkerConfig {
+		ProblemMarkerConfig(final int severity) {
+			super(PROBLEMMARKER_TYPE, IMarker.PRIORITY_NORMAL, severity);
+		}
+	}
+
+	private static final class TaskMarkerConfig extends AbstractMarkerConfig {
+		TaskMarkerConfig(final int priority) {
+			super(TASKMARKER_TYPE, priority, IMarker.SEVERITY_INFO);
+		}
+	}
+
+	private static final String TEXTMARKER_TYPE = "org.eclipse.tm4e.ui.textmarker";
+	private static final String PROBLEMMARKER_TYPE = "org.eclipse.tm4e.ui.problemmarker";
+	private static final String TASKMARKER_TYPE = "org.eclipse.tm4e.ui.taskmarker";
+
+	private static final Map<String, AbstractMarkerConfig> MARKERCONIG_BY_TAG = Map.of(
+			// problem markers:
+			"BUG", new ProblemMarkerConfig(IMarker.SEVERITY_ERROR),
+			"NOTE", new ProblemMarkerConfig(IMarker.SEVERITY_INFO),
+			// task markers:
+			"FIXME", new TaskMarkerConfig(IMarker.PRIORITY_HIGH),
+			"HACK", new TaskMarkerConfig(IMarker.PRIORITY_NORMAL),
+			"TODO", new TaskMarkerConfig(IMarker.PRIORITY_NORMAL),
+			"XXX", new TaskMarkerConfig(IMarker.PRIORITY_NORMAL));
+
+	private static final Pattern TAG_SELECTOR_PATTERN = Pattern.compile(
+			"\\b(" + MARKERCONIG_BY_TAG.keySet().stream().collect(Collectors.joining("|")) + ")\\b");
+
+	private static int getLineNumber(final IMarker marker) {
+		try {
+			final var lineNumberAttr = marker.getAttribute(IMarker.LINE_NUMBER);
+			if (lineNumberAttr instanceof final Integer lineNumber)
+				return lineNumber;
+		} catch (final CoreException ex) {
+			TMUIPlugin.log(new Status(IStatus.ERROR, TMUIPlugin.PLUGIN_ID, ex.getMessage(), ex));
+		}
+		return -1;
+	}
+
+	public static void updateTextMarkers(final ModelTokensChangedEvent event) {
```

Comment:
Move this staic public method before privatif methods. Add j1vadoc for this method

Reply:
Going through the code base I actually haven't yet seen any continuous convention how methods are ordered, why do you want to have the method order changed here? Is this a best practice/convention that is documented anywhere?

Reply:
IMHO public code is more important than private code. When I open a Java file, I like to see on the top the public method that I can use it although private code is less important.